titan-compiler/syntax_errors.lua

```diff
@@ -26,6 +27,108 @@ local errors = {
 
     { label = "MalformedEscape_x",
         msg = "\\x escape sequences must have exactly two hexadecimal digits." },
+    
+    { label = "NameFunc",
+        msg = "Expected a function name after 'function'." },
+    
+    { label = "OParPList",
+        msg = "Expected '(' for the parameter list." },
+    
+    { label = "CParPList",
+        msg = "Expected ')' to close the parameter list." },
+    
+    { label = "ColonFunc",
+        msg = "Expected ':' after the parameter list." },
+    
+    { label = "TypeFunc",
+        msg = "Expected a type in function declaration." },
+    
+    { label = "EndFunc",
+        msg = "Expected 'end' to close the function body." },
+    
+    { label = "AssignVar",
+        msg = "Expected '=' after variable declaration." },
+    
+    { label = "ExpVarDec",
+        msg = "Expected an expression to initialize variable." },
+    
+    { label = "NameRecord",
+        msg = "Expected a record name after 'record'." },
+    
+    { label = "EndRecord",
+        msg = "Expected 'end' to close the record." },
+    
+    { label = "FieldRecord",
+        msg = "Expected a field in record declaration." },
+    
+    { label = "NameImport",
+        msg = "Expected a name after 'local'." },
+   
+   --this label is not thrown in rule 'import' because rule 'toplevelvar'
+   --matches an invalid input like "local bola import"
+   { label = "AssignImport",
+        msg = "Expected '='." },
+   
+   --this label is not thrown in rule 'import' because rule 'toplevelvar'
+   --matches an input like "local bola = X", given that X is a valid expression,
+   --or throws a label when X is not a valid expression
+   { label = "ImportImport",
+        msg = "Expected 'import' keyword." },
```

Comment:
Perhaps a way to avoid this issue would be to syntactically allow assignment statements at the toplevel and use a `=>` capture to tell which of them are import declarations, which are toplevel variables and which should be considered syntax errors.